mm2src/mm2_main/src/rpc/wc_commands/new_connection.rs

```diff
@@ -0,0 +1,32 @@
+use kdf_walletconnect::WalletConnectCtx;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
+use serde::{Deserialize, Serialize};
+
+use super::WalletConnectRpcError;
+
+#[derive(Debug, PartialEq, Serialize)]
+pub struct CreateConnectionResponse {
+    pub url: String,
+}
+
+#[derive(Deserialize)]
+pub struct NewConnectionRequest {
+    required_namespaces: serde_json::Value,
+    optional_namespaces: Option<serde_json::Value>,
+}
+
+/// `new_connection` RPC command implementation.
+pub async fn new_connection(
+    ctx: MmArc,
+    req: NewConnectionRequest,
+) -> MmResult<CreateConnectionResponse, WalletConnectRpcError> {
+    let ctx =
```

Comment:
let's name this something other than ctx, similar for others. It should be something like `wc_ctx`